rest_server/main.go

```diff
@@ -0,0 +1,127 @@
+		fmt.Printf("%v %v %v %v REST Server | %s\n", time.Now().UTC().Format("2006-01-02 15:04:05.999999"),
+			SeverityLevelToString(FATAL), unix.Getpid(), unix.Gettid(), err)
+		os.Exit(2)
+	}
+	siodbLoggerPool.ConfigGinLogger()
+	siodbLoggerPool.Info("Logger Pool initialized successfully")
+
+	// Create REST Server config
+	if err = restServerConfig.ParseAndValidateConfiguration(siodbConfigFile); err != nil {
+		siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "cannot parse the rest configuration properly", err)
+	}
+	siodbLoggerPool.Info("REST server config initialized successfully")
+
+	// Connection pool to IOMgr
+	IOMgrCPool, err = CreateIOMgrConnPool(siodbConfigFile, IOMgrCPoolMinConn, IOMgrCPoolMaxConn)
+	if err != nil {
+		siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "Cannot create connection pool: %v", err)
+	}
+	siodbLoggerPool.Info("IOMgr connection pool initialized successfully")
+
+	// Start routers
+	if restServerConfig.Ipv4HTTPPort != 0 {
+		go func() {
+			var restWorker RestWorker
+			if err := restWorker.CreateRouter(restServerConfig.Ipv4HTTPPort); err != nil {
+				siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "Cannot create route: %v", err)
+			}
+			if err := restWorker.StartHTTPRouter(); err != nil {
+				siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "Cannot start router: %v", err)
+			}
+		}()
+	}
+	if restServerConfig.Ipv4HTTPSPort != 0 {
+		go func() {
+			var restWorker RestWorker
+			if err := restWorker.CreateRouter(restServerConfig.Ipv4HTTPSPort); err != nil {
+				siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "Cannot create route: %v", err)
+			}
+			if err := restWorker.StartHTTPSRouter(restServerConfig.TLSCertificate, restServerConfig.TLSPrivateKey); err != nil {
+				siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "Cannot start router: %v", err)
+			}
+		}()
+	}
+	if restServerConfig.Ipv6HTTPPort != 0 {
+		go func() {
+			var restWorker RestWorker
+			if err := restWorker.CreateRouter(restServerConfig.Ipv6HTTPPort); err != nil {
+				siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "Cannot create route: %v", err)
+			}
+			if err := restWorker.StartHTTPRouter(); err != nil {
+				siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "Cannot start router: %v", err)
+			}
+		}()
+	}
+	if restServerConfig.Ipv6HTTPSPort != 0 {
+		go func() {
+			var restWorker RestWorker
+			if err := restWorker.CreateRouter(restServerConfig.Ipv6HTTPSPort); err != nil {
+				siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "Cannot create route: %v", err)
+			}
+			if err := restWorker.StartHTTPSRouter(restServerConfig.TLSCertificate, restServerConfig.TLSPrivateKey); err != nil {
+				siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "Cannot start router: %v", err)
+			}
+		}()
+	}
+
+	sig := <-sigs
+	siodbLoggerPool.Info("%v signal received, terminating... ", sig)
+	err = IOMgrCPool.CloseAllConnections()
+	if err != nil {
+		siodbLoggerPool.FatalAndExit(FATAL_INIT_ERROR, "Cannot close connection pool: %v", err)
+	}
+}
```

Comment:
at this point you should output "INFO Logging stopped" and close all non-console log channels.